examples/SimilarityScoring.ipynb

```diff
@@ -0,0 +1,215 @@
+    "import torch\n",
+    "import numpy as np\n",
+    "from torch import nn\n",
+    "import urllib.request\n",
+    "\n",
+    "\n",
+    "# Test images to be explained\n",
+    "urllib.request.urlretrieve('https://farm1.staticflickr.com/74/202734059_fcce636dcd_z.jpg', \"catdog.jpg\")\n",
+    "test_image1_path = 'catdog.jpg'\n",
+    "\n",
+    "urllib.request.urlretrieve('https://farm5.staticflickr.com/4089/4990202073_725f035a15_z.jpg', \"dog.jpg\")\n",
+    "test_image2_path = 'dog.jpg'\n",
+    "\n",
+    "# Constructing xaitk implementations for SimilarityBased Saliency genration with\n",
+    "# sliding windows.\n",
+    "slid_algo = SlidingWindowPerturb(window_size=(24, 24), stride=(5, 5), threads=4)\n",
+    "similarity_alg = SimilarityScoring()\n",
+    "\n",
+    "\n",
+    "# Creating an example deep feature extractor to describe images\n",
+    "model = models.resnet18(pretrained=True)\n",
+    "model = nn.Sequential(*list(model.children())[:-1])\n",
+    "model = model.eval()\n",
+    "model = model.cuda()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Input image loader to match feature extractor expectation\n",
+    "model_input_size = 224\n",
+    "model_loader = transforms.Compose([\n",
+    "    transforms.Resize(model_input_size), \n",
+    "    transforms.ToTensor(),\n",
+    "    transforms.Normalize(\n",
+    "        mean=[0.485, 0.456, 0.406],\n",
+    "        std=[0.229, 0.224, 0.225]\n",
+    "    ),\n",
+    "])\n",
+    "\n",
+    "def image_loader(image_name):\n",
+    "    try:\n",
+    "        image = Image.open(image_name)\n",
+    "    except AttributeError:\n",
+    "        # If Image already open\n",
+    "        image = image_name\n",
+    "    image = model_loader(image).float()\n",
+    "    image = Variable(image, requires_grad=False)\n",
+    "    return image.cuda().unsqueeze(0)\n",
+    "\n",
+    "def predict_with_model(model, image_tensor):\n",
+    "    featureVec = model(image_tensor)\n",
+    "    return featureVec.cpu().detach().numpy().squeeze()\n",
+    "\n",
+    "def compute_prtb_feats(p_image):\n",
+    "    # Use created sliding window based perturbation algorithm to perturb test image\n",
+    "    pert_img = []\n",
+    "    pertmask = []\n",
+    "    prtbd_feat = []\n",
+    "    for i, (img, mask) in enumerate(slid_algo(p_image)):\n",
+    "        pertmask.append(mask)\n",
+    "        pert_img.append(img)\n",
+    "    # Compute feature vectors on perturbed version of the test image\n",
+    "    prtbd_feat = np.array([predict_with_model(model, image_loader(ptrbed)) for ptrbed in pert_img])\n",
+    "    return np.array(pertmask), prtbd_feat\n",
+    "\n",
+    "# Test images containing cat and dog.\n",
+    "test_image1 = Image.open(test_image1_path).resize((imsize, imsize))\n",
+    "test_image2 = Image.open(test_image2_path).resize((imsize, imsize))\n",
```

Comment:
I get errors running this notebook saying that `imsize` is not defined here.

Reply:
Good catch, corrected it to track `model_input_size`